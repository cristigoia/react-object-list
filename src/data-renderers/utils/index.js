export {
  getValueFromAccessor,
  handleRowClick,
} from './functions'
