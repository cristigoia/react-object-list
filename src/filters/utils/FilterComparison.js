import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'

class FilterComparison extends React.Component {
  static propTypes = {
    options: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string,
      value: Select.propTypes.value,
    })),
    value: Select.propTypes.value,
    onChange: PropTypes.func.isRequired,
  }

  render() {
    const {options, value} = this.props
    return (
      <div className="objectlist-current-filter__filter-comparison">
        <Select
          options={options}
          value={value}
          onChange={newValue => this.props.onChange(newValue.value)}
          clearable={false}
        />
      </div>
    )
  }
}

export default FilterComparison
