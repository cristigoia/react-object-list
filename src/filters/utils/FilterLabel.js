import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render label for filters
 */
class FilterLabel extends React.Component {
  static propTypes = {
    /** Text to render in the label */
    label: PropTypes.string.isRequired,
  }
  render() {
    return (
      <label className="objectlist-current-filter__filter-name">{this.props.label}:</label>
    )
  }
}

export default FilterLabel
