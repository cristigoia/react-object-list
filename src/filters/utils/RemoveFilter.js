import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render button used to remove a filter
 */
class RemoveFilter extends React.Component {
  static propTypes = {
    /** Function to be called to remove filter */
    onClick: PropTypes.func.isRequired,
  }

  render() {
    return (
      <button className="objectlist-button objectlist-button--delete objectlist-button--icon objectlist-button--borderless" onClick={this.props.onClick} title="Remove Filter" />
    )
  }
}

export default RemoveFilter
