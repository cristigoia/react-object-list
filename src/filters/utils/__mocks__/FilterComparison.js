import React from 'react'

const MockFilterComparison = props => React.createElement('FilterComparison', props)
const RealFilterComparison = require.requireActual('../FilterComparison').default
/* eslint-disable-next-line */
MockFilterComparison.propTypes = RealFilterComparison.propTypes

export default MockFilterComparison
