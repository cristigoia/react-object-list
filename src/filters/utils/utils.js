const possibleQueryStringLookupExpressions = ['exact', 'lt', 'gt', 'gte', 'lte']

export {
  possibleQueryStringLookupExpressions,
}
