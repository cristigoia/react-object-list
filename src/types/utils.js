function floatPrecision(floatValue, precision) {
  floatValue = parseFloat(floatValue)
  if (isNaN(floatValue)) { return parseFloat('0').toFixed(precision) } else {
    const power = Math.pow(10, precision)
    floatValue = (Math.round(floatValue * power) / power).toFixed(precision)
    return floatValue.toString()
  }
}

function commaNumber(numberString) {
  const parts = numberString.toString().split('.')
  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',')
  return parts.join('.')
}
/**
 * Converts a float to stringed currency representation
 * @param  {number} floatNumber Amount to convert
 * @return {string}             Dollar string
 */
export function currency(floatNumber) {
  return `${floatNumber < 0 ? '-' : ''}$${commaNumber(floatPrecision(Math.abs(floatNumber), 2))}`
}
