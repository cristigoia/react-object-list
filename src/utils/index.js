export * from './constants'
export * from './proptypes'
export * from './functions'
export loadingSpinner from './loading'
export ErrorMessage from './ErrorMessage'
