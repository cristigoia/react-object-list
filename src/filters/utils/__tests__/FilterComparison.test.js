import React from 'react'
import { snapshotTest } from 'utils/tests'
import FilterComparison from '../FilterComparison'

jest.mock('react-select')

describe('FilterComparison', () => {
  const baseProps = {
    onChange: jest.fn(),
    value: 'test',
    options: [{value: 'test', label: 'Test This'}],
  }
  describe('Snapshots', () => {
    it('renders default', () => {
      snapshotTest(<FilterComparison {...baseProps} />)
    })
  })
})
