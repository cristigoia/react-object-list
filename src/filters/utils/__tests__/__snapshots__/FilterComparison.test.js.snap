// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterComparison Snapshots renders default 1`] = `
<div
  className="objectlist-current-filter__filter-comparison"
>
  <Select
    clearable={false}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Test This",
          "value": "test",
        },
      ]
    }
    value="test"
  />
</div>
`;
